### ISSUE TITLE
Click doesn't work as expected in non-Headless

### ISSUE THREAD

**Issue by alixaxel** [puppeteer/puppeteer#1126]
**Environment:**

* Puppeteer version: 0.12.0 / 0.13.0-alpha
* Platform / OS version: Ubuntu 17.04
* URLs: [Test in PR](https://github.com/GoogleChrome/puppeteer/pull/1125) / `https://www.htmlbin.net/login`

**What steps will reproduce the problem?**

```javascript
const puppeteer = require('puppeteer');
let browser = null;

(async () => {
  browser = await puppeteer.launch({
    headless: true,
  });

  const page = await browser.newPage();
  await page.goto('https://www.htmlbin.net/login');
  await page.click('.checkbox input');
  await page.screenshot({path: 'headless_true.png'});
  await browser.close();

  browser = await puppeteer.launch({
    headless: false,
  })

  const page = await browser.newPage();
  await page.goto('https://www.htmlbin.net/login');
  await page.click('.checkbox input');
  await page.screenshot({path: 'headless_false.png'});
  await browser.close();
})();
```

**What is the expected result?**

Both screenshots should show the checkbox in the checked state.

**What happens instead?**

Checkbox is only shown in checked state when browser is launched with `headless: true`.

No mouse/click events are triggered for `headless: false` either, I've added more information in #1125.

**Comment by akob** [comment]
So when it will be fixed?

**Comment by phawxby** [comment]
Confirmed in V0.12.0 with anchors too.

**Comment by hiitsu** [comment]
yes, i have this too in 0.12.0 both headless and with ui, node 8.6.0 + OSX,  tests hanging in the step next to .click call when waiting for next thing to appear, if manually press it while test is in progress it'll finish normally

**Comment by JoelEinbinder** [comment]
This should be fixed by #1299
